nise/generators/oci/oci_generator.py

```diff
@@ -90,64 +92,28 @@
 class OCIGenerator(AbstractGenerator):
     """Defines a abstract class for generators."""
 
-    OCI_REGIONS_TO_DOMAIN = [
-        {"region": "ap-sydney-1", "domain": 1},
-        {"region": "ap-melbourne-1", "domain": 1},
-        {"region": "sa-saopaulo-1", "domain": 1},
-        {"region": "sa-vinhedo-1", "domain": 1},
-        {"region": "ca-montreal-1", "domain": 1},
-        {"region": "ca-toronto-1", "domain": 1},
-        {"region": "sa-santiago-1", "domain": 1},
-        {"region": "eu-marseille-1", "domain": 1},
-        {"region": "eu-frankfurt-1", "domain": 3},
-        {"region": "ap-hyderabad-1", "domain": 1},
-        {"region": "ap-mumbai-1", "domain": 1},
-        {"region": "il-jerusalem-1", "domain": 1},
-        {"region": "eu-milan-1", "domain": 1},
-        {"region": "ap-osaka-1", "domain": 1},
-        {"region": "ap-tokyo-1", "domain": 1},
-        {"region": "eu-amsterdam-1", "domain": 1},
-        {"region": "me-jeddah-1", "domain": 1},
-        {"region": "ap-singapore-1", "domain": 1},
-        {"region": "af-johannesburg-1", "domain": 1},
-        {"region": "ap-seoul-1", "domain": 1},
-        {"region": "ap-chuncheon-1", "domain": 1},
-        {"region": "eu-stockholm-1", "domain": 1},
-        {"region": "eu-zurich-1", "domain": 1},
-        {"region": "me-abudhabi-1", "domain": 1},
-        {"region": "me-dubai-1", "domain": 1},
-        {"region": "uk-london-1", "domain": 1},
-        {"region": "uk-cardiff-1", "domain": 1},
-        {"region": "us-ashburn-1", "domain": 3},
-        {"region": "us-phoenix-1", "domain": 3},
-        {"region": "us-sanjose-1", "domain": 1},
-    ]
-
     def __init__(self, start_date, end_date, currency, attributes=None):
         """Initialize the generator."""
         super().__init__(start_date, end_date)
+        if attributes is None or not isinstance(attributes, dict):
+            attributes = {}
         self.currency = currency
-        self.tenant_id = (
-            attributes.get("tenant_id")
-            if attributes
-            else f"ocid1.tenancy.oc1..{self.fake.pystr(min_chars=20, max_chars=50)}"
-        )
+        self.tenant_id = attributes.get("tenant_id", OCIReportConstantColumns.tenant_id)
         self.reference_no = self._get_reference_num()
         self.compartment_id = self.tenant_id
-        self.compartment_name = (
-            attributes.get("compartment_name")
-            if attributes and attributes.get("compartment_name")
-            else self.fake.name().replace(" ", "").lower()
-        )
-        self.region_to_domain = choice(self.OCI_REGIONS_TO_DOMAIN)
+        self.compartment_name = attributes.get("compartment_name", OCIReportConstantColumns.compartment_name)
+        self.region_to_domain = OCIReportConstantColumns().oci_regions_to_domain
```

Comment:
I don't see much value in creating an `OCIReportConstantColumns` object just to get a random region object. Calling  `random_region_domain` directly would be more efficient.

If there was an instance of `OCIReportConstantColumns`, this would make more sense.

```suggestion
        self.region_to_domain = random_region_domain()
```

Reply:
I'm thinking it would make sense to have an instance of `OCIReportConstantColumns` attached to this class. That seems the most coherent interface to the constants. Instances of data classes are meant to be interacted with, not the classes themselves.